pkg/client/event.go

```diff
@@ -15,14 +15,16 @@ import (
 
 type pushOpt struct {
 	additionalMetadata map[string]string
+	priority           *int32
+	scope              *string
 }
 
 type PushOpFunc func(*pushOpt) error
 
 type BulkPushOpFunc func(*eventcontracts.BulkPushEventRequest) error
 
 type EventClient interface {
-	Push(ctx context.Context, eventKey string, payload interface{}, options ...PushOpFunc) error
+	Push(ctx context.Context, eventKey string, payload interface{}, priority *int32, scope *string, options ...PushOpFunc) error
```

Comment:
this is a breaking change, but I don't know what the convention here is - can these go after `options`?

Reply:
We can add these as `PushOpFunc` options (`WithPriority`, `WithScope`) to prevent the breaking change